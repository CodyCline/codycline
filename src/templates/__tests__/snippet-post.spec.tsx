import React from "react"
import renderer from "react-test-renderer";
import SnippetPostTemplate from "../snippet-post";


describe(`Snippets`, () => {
    it(`renders GraphQL data correctly`, () => {
        const pageContext = {
            next: null,
            previous: {
                fields: { slug: "/snippet/test-snippet" },
                frontmatter: { title: "Title" },
                slug: "/snippet/test-snippet/"
            }
        }
        const data = {
            site: {
                siteMetadata: {
                    title: "Default Starter",
                    social: {
                        github: "test_user"
                    }
                }
            },
            mdx: {
                id: "uuid",
                frontmatter: {
                    tags: [
                        "kubernetes",
                    ],
                    title: "Test Article",
                    date: "1970-01-01",
                    description: "Test Article",
                },
                body: "function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }\n\nfunction _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }\n\nfunction _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }\n\n/* @jsx mdx */\nvar _frontmatter = {\n  \"date\": \"1970-01-01\",\n  \"tags\": [\"rust\"],\n  \"title\": \"Markdown Test\",\n  \"description\": \"Build a blockchain in rust\"\n};\nvar layoutProps = {\n  _frontmatter: _frontmatter\n};\nvar MDXLayout = \"wrapper\";\nreturn function MDXContent(_ref) {\n  var components = _ref.components,\n      props = _objectWithoutProperties(_ref, [\"components\"]);\n\n  return mdx(MDXLayout, _extends({}, layoutProps, props, {\n    components: components,\n    mdxType: \"MDXLayout\"\n  }), mdx(\"p\", null, \"Hello world\"));\n}\n;\nMDXContent.isMDXComponent = true;" 
            }
        }
        const tree = renderer.create(<SnippetPostTemplate data={data} pageContext={pageContext} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})