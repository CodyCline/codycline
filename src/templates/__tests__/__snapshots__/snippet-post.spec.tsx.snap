// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippets renders GraphQL data correctly 1`] = `
Array [
  <nav
    className="navbar"
  >
    <a
      className="nav__item home__icon"
      href="/"
    >
      <img
        alt="logo.png"
        src="test-file-stub"
        style={
          Object {
            "height": "50px",
            "width": "70px",
          }
        }
      />
    </a>
    <a
      className="nav__item"
      href="/blog"
    >
       Blog 
    </a>
    <a
      className="nav__item"
      href="/snippets"
    >
      Snippets
    </a>
    <a
      className="nav__item"
      href="/projects"
    >
      Projects
    </a>
    <a
      className="nav__item controls"
      href="#"
    />
  </nav>,
  <article
    className="article__container"
    id="uuid"
  >
    <h1
      className="article__large-header"
    >
      Test Article
    </h1>
    <h3>
      1970-01-01
    </h3>
    <hr
      className="divider"
    />
    <section
      className="article__markdown-section"
    >
      <p>
        Hello world
      </p>
    </section>
    <div
      className="article__tags"
    >
      <a
        className="meta"
        href="/meta/kubernetes"
      >
        <svg
          height={24}
          viewBox="0 0 1024 1024"
          width={24}
        >
          <title>
            kubernetes
          </title>
          <path
            d="M523.679 14.673c-9.586 0.496-18.498 2.882-26.523 6.787l0.401-0.176-356.441 170.311c-18.494 8.976-32.077 25.494-36.788 45.402l-0.091 0.458-87.937 382.563c-1.088 4.524-1.712 9.719-1.712 15.058 0 13.671 4.091 26.394 11.126 37.030l-0.159-0.257c1.351 2.075 2.626 3.829 3.981 5.517l-0.106-0.137 246.715 306.752c12.596 15.561 31.774 25.437 53.277 25.437 0.005 0 0.009-0 0.014-0l395.645-0.091c0.015 0 0.032 0 0.050 0 21.483 0 40.645-9.857 53.138-25.261l0.103-0.131 246.623-306.798c9.274-11.449 14.884-26.158 14.884-42.166 0-5.334-0.623-10.524-1.801-15.503l0.091 0.458-88.073-382.563c-4.802-20.366-18.386-36.884-36.478-45.684l-0.402-0.176-356.487-170.22c-8.646-4.226-18.822-6.698-29.58-6.698-1.221 0-2.434 0.032-3.639 0.095l0.168-0.007z"
            fill="rgb(50, 108, 229)"
            opacity={1}
          />
          <path
            d="M527.211 144.864c-11.782 0.001-21.336 10.614-21.335 23.705 0 0.201 0.041 0.393 0.046 0.593-0.017 1.779-0.103 3.922-0.046 5.47 0.281 7.551 1.927 13.33 2.918 20.286 1.795 14.889 3.299 27.231 2.371 38.703-0.903 4.326-4.089 8.282-6.929 11.032l-0.501 9.026c-12.803 1.061-25.692 3.003-38.566 5.926-55.396 12.578-103.091 41.113-139.404 79.64-2.356-1.607-6.478-4.565-7.704-5.47-3.809 0.514-7.659 1.69-12.673-1.231-9.548-6.427-18.243-15.298-28.765-25.984-4.821-5.112-8.313-9.979-14.041-14.907-1.301-1.119-3.286-2.632-4.741-3.784-4.478-3.57-9.76-5.432-14.861-5.607-6.558-0.225-12.872 2.34-17.004 7.522-7.345 9.213-4.994 23.294 5.242 31.455 0.104 0.083 0.214 0.147 0.319 0.228 1.407 1.14 3.129 2.601 4.422 3.556 6.079 4.488 11.631 6.785 17.688 10.348 12.76 7.88 23.338 14.414 31.728 22.292 3.276 3.493 3.849 9.647 4.285 12.308l6.838 6.109c-36.606 55.089-53.547 123.134-43.535 192.467l-8.935 2.598c-2.355 3.041-5.683 7.826-9.163 9.254-10.977 3.458-23.332 4.727-38.247 6.291-7.003 0.582-13.045 0.235-20.468 1.641-1.634 0.31-3.911 0.903-5.698 1.322-0.062 0.013-0.12 0.032-0.182 0.046-0.097 0.023-0.225 0.070-0.319 0.091-12.575 3.038-20.653 14.596-18.052 25.984 2.601 11.391 14.884 18.317 27.534 15.591 0.091-0.021 0.224-0.024 0.319-0.046 0.143-0.033 0.269-0.102 0.41-0.137 1.763-0.387 3.973-0.818 5.516-1.231 7.299-1.954 12.585-4.826 19.146-7.339 14.116-5.063 25.808-9.293 37.199-10.941 4.757-0.373 9.77 2.935 12.263 4.331l9.3-1.596c21.4 66.349 66.249 119.977 123.038 153.627l-3.875 9.3c1.397 3.611 2.937 8.497 1.897 12.063-4.141 10.738-11.234 22.073-19.311 34.709-3.911 5.838-7.913 10.368-11.442 17.049-0.844 1.599-1.92 4.054-2.735 5.744-5.483 11.732-1.461 25.244 9.072 30.315 10.599 5.103 23.755-0.279 29.449-12.035 0.009-0.017 0.037-0.029 0.046-0.046 0.006-0.013-0.006-0.033 0-0.046 0.811-1.667 1.96-3.858 2.644-5.425 3.023-6.925 4.029-12.859 6.154-19.557 5.645-14.179 8.746-29.057 16.517-38.328 2.128-2.539 5.597-3.515 9.194-4.478l4.832-8.753c49.508 19.003 104.925 24.102 160.282 11.533 12.629-2.867 24.82-6.578 36.606-11.032 1.358 2.409 3.882 7.040 4.559 8.206 3.655 1.189 7.645 1.803 10.895 6.61 5.814 9.933 9.79 21.683 14.633 35.877 2.126 6.697 3.177 12.632 6.2 19.557 0.689 1.578 1.832 3.8 2.644 5.47 5.682 11.794 18.88 17.194 29.495 12.080 10.532-5.073 14.558-18.585 9.072-30.315-0.815-1.689-1.936-4.145-2.781-5.744-3.529-6.681-7.531-11.166-11.442-17.004-8.078-12.636-14.777-23.133-18.918-33.871-1.732-5.538 0.292-8.983 1.641-12.582-0.808-0.926-2.537-6.156-3.556-8.616 59.018-34.847 102.549-90.475 122.993-154.721 2.761 0.434 7.559 1.283 9.117 1.596 3.209-2.116 6.159-4.877 11.944-4.422 11.391 1.647 23.082 5.878 37.199 10.941 6.562 2.514 11.848 5.431 19.146 7.385 1.543 0.413 3.753 0.798 5.516 1.185 0.142 0.035 0.267 0.104 0.41 0.137 0.095 0.021 0.228 0.025 0.319 0.046 12.651 2.723 24.936-4.199 27.534-15.591 2.598-11.389-5.477-22.949-18.052-25.984-1.829-0.416-4.423-1.122-6.2-1.459-7.424-1.406-13.466-1.059-20.468-1.641-14.915-1.563-27.27-2.834-38.247-6.291-4.476-1.736-7.66-7.062-9.209-9.254l-8.616-2.507c4.467-32.318 3.263-65.953-4.467-99.607-7.802-33.967-21.59-65.033-39.979-92.404 2.21-2.009 6.384-5.705 7.567-6.792 0.346-3.828 0.049-7.842 4.012-12.080 8.39-7.878 18.969-14.411 31.728-22.292 6.056-3.563 11.655-5.86 17.733-10.348 1.375-1.015 3.251-2.622 4.695-3.784 10.234-8.164 12.59-22.244 5.242-31.455s-21.586-10.078-31.819-1.915c-1.457 1.154-3.433 2.659-4.741 3.784-5.728 4.928-9.265 9.795-14.086 14.907-10.521 10.687-19.218 19.603-28.765 26.030-4.137 2.409-10.197 1.575-12.947 1.413l-8.114 5.789c-46.27-48.52-109.269-79.54-177.104-85.566-0.19-2.843-0.438-7.981-0.501-9.528-2.777-2.657-6.132-4.926-6.975-10.667-0.928-11.472 0.621-23.814 2.416-38.703 0.991-6.957 2.637-12.735 2.918-20.286 0.064-1.716-0.039-4.207-0.046-6.063-0.001-13.091-9.552-23.706-21.335-23.705zM500.497 310.344l-6.337 111.915-0.456 0.228c-0.425 10.012-8.665 18.007-18.782 18.007-4.144 0-7.969-1.331-11.078-3.601l-0.182 0.091-91.766-65.052c28.203-27.733 64.278-48.228 105.852-57.667 7.594-1.724 15.185-3.004 22.748-3.92zM553.97 310.344c48.538 5.97 93.427 27.948 127.825 61.633l-91.173 64.642-0.319-0.137c-8.092 5.91-19.494 4.444-25.802-3.465-2.584-3.24-3.94-7.049-4.103-10.895l-0.091-0.046zM338.619 413.734l83.788 74.944-0.091 0.456c7.563 6.575 8.678 17.984 2.371 25.893-2.584 3.24-6.042 5.413-9.756 6.428l-0.091 0.365-107.402 30.999c-5.466-49.985 6.314-98.574 31.181-139.085zM715.21 413.78c12.449 20.179 21.877 42.716 27.489 67.149 5.545 24.14 6.936 48.237 4.65 71.525l-107.949-31.090-0.091-0.456c-9.667-2.642-15.607-12.475-13.357-22.337 0.922-4.040 3.067-7.458 5.972-9.983l-0.046-0.228 83.332-74.58zM510.025 494.468h34.327l21.335 26.668-7.659 33.278-30.817 14.816-30.908-14.861-7.659-33.278zM620.071 585.732c1.459-0.074 2.911 0.058 4.331 0.319l0.182-0.228 111.095 18.782c-16.259 45.679-47.37 85.251-88.939 111.733l-43.125-104.165 0.137-0.182c-3.961-9.205 0.003-19.999 9.117-24.389 2.334-1.124 4.771-1.746 7.203-1.869zM433.485 586.188c8.478 0.119 16.082 6.003 18.052 14.633 0.922 4.040 0.473 8.043-1.048 11.579l0.319 0.41-42.669 103.117c-39.893-25.599-71.668-63.929-88.666-110.958l110.137-18.69 0.182 0.228c1.232-0.227 2.481-0.336 3.693-0.319zM526.527 631.364c2.953-0.109 5.95 0.497 8.798 1.869 3.734 1.798 6.618 4.629 8.434 8.023h0.41l54.294 98.102c-7.046 2.362-14.29 4.381-21.699 6.063-41.523 9.428-82.915 6.571-120.394-6.2l54.157-97.92h0.091c3.25-6.075 9.413-9.699 15.91-9.938z"
            fill="rgb(255, 255, 255)"
            opacity={1}
          />
        </svg>
        <span
          className="meta__tag"
        >
          kubernetes
        </span>
      </a>
    </div>
    <nav
      className="article__navigation"
    >
      <li>
        <a
          href="/snippet/test-snippet"
          rel="prev"
        >
          ‚Üê 
          Title
        </a>
      </li>
      <li />
    </nav>
  </article>,
  <footer
    className="footer"
  >
    <li
      className="footer__item"
    >
      2021
    </li>
    <li
      className="footer__item"
    />
  </footer>,
]
`;
